include/seqan/arg_parse/arg_parse_version_check.h

```diff
@@ -0,0 +1,459 @@
+    strSplit(set, str, EqualsChar<'.'>(), false);
+    for (auto & num : set)
+        appendValue(numbers, lexicalCast<int>(num));
+    return numbers;
+}
+
+// ----------------------------------------------------------------------------
+// Function _readVersionString()
+// ----------------------------------------------------------------------------
+
+inline std::string _readVersionString(std::string const & version_file)
+{
+    std::ifstream myfile;
+    myfile.open(version_file.c_str());
+    std::string line;
+    if (myfile.is_open())
+    {
+        std::getline(myfile,line); // get first line which should only contain the version number
+        if (!(std::regex_match(line, std::regex("^[[:digit:]]+\\.[[:digit:]]+\\.[[:digit:]]+$"))))
+        {
+            line.clear();
+        }
+        if (line == VersionControlTags_<>::UNREGISTERED_APP)
+        {
+            std::cerr << "[SEQAN INFO] :: Thank you for using SeqAn!\n"
+                      << "[SEQAN INFO] :: You might want to regsiter you app for support and version check features?!\n"
+                      << "[SEQAN INFO] :: Just send us an email to seqan@team.fu-berlin.de with your app name and version number.\n"
+                      << "[SEQAN INFO] :: If you don't want to recieve this message anymore set --version_check 2\n\n";
+            line.clear();
+        }
+        myfile.close();
+    }
+
+    return line; // line is an empty string on failure
+}
+
+// ----------------------------------------------------------------------------
+// Function _callServer()
+// ----------------------------------------------------------------------------
+
+inline void _callServer(VersionCheck const me, std::promise<bool> prom)
+{
+    // update timestamp
+    std::string timestamp_filename = me._path + "/" + me._name + ".timestamp";
+    std::ofstream timestamp_file(timestamp_filename.c_str());
+    if (timestamp_file.is_open())
+    {
+        timestamp_file << std::chrono::duration_cast<std::chrono::seconds>(std::chrono::system_clock::now().time_since_epoch()).count();
+        timestamp_file.close();
+    }
+
+    // system call
+    // http response is stored in a file '.config/seqan/{app_name}_version'
+    if (system(me._command.c_str()))
+        prom.set_value(false);
+    else
+        prom.set_value(true);
+}
+
+// ----------------------------------------------------------------------------
+// Function checkForNewerVersion()
+// ----------------------------------------------------------------------------
+
+inline void _checkForNewerVersion(VersionCheck & me, std::promise<bool> prom)
+{
+    if (!_checkWritability(me._path))
+    {
+#if defined(STDLIB_VS)
+        TCHAR tmp_path [MAX_PATH];
+        if (GetTempPath(MAX_PATH, tmp_path) != 0)
+        {
+            me._path = tmp_path;
+        }
+        else
+        { //GetTempPath() returns 0 on failure
+            prom.set_value(false);
+            return;
+        }
+# else // unix
+        me._path = "/tmp";
```

Comment:
I can only find the SEQAN_DEFAULT_TMPDIR defined inside some apps 
(e.g. ```./apps/micro_razers/micro_razers.cpp:32:    #define SEQAN_DEFAULT_TMPDIR "C:\\TEMP\\"```)
but nowhere globally. I then looked into the seqan macro SEQAN_TEMP_FILENAME, since it needs the temp folder of course and, as I see it, it does nothing differently from what I did:
https://github.com/seqan/seqan/blob/master/include/seqan/basic/debug_test_system.h#L681